Tests/UnitTests/Plugins/Geant4/Geant4DetectorSurfaceFactoryTests.cpp

```diff
@@ -94,4 +97,97 @@ BOOST_AUTO_TEST_CASE(Geant4DetecturSurfaceFactory_Cylinder) {
+  auto vol1L = new G4LogicalVolume(vol1S, nullptr, "Volume1");
+
+  G4Transform3D transformVol1(CLHEP::HepRotationX(M_PI / 4),
+                              G4ThreeVector(20, 0, 0));
+
+  [[maybe_unused]] auto vol1PV = new G4PVPlacement(
+      transformVol1, vol1L, "Volume1", worldLV, false, 0, false);
+
+  auto vol2S = new G4Box("volume2", 25, 10, 50);
+  auto vol2L = new G4LogicalVolume(vol2S, nullptr, "Volume2");
+
+  G4Transform3D transformVol2(CLHEP::HepRotationY(M_PI / 6),
+                              G4ThreeVector(0, 100, 20));
+
+  [[maybe_unused]] auto vol2PV = new G4PVPlacement(
+      transformVol2, vol2L, "Volume2", vol1L, false, 0, false);
+
+  auto vol3S = new G4Box("volume3", 25, 10, 50);
+  auto vol3L = new G4LogicalVolume(vol3S, nullptr, "Volume3");
+
+  G4Transform3D transformVol3(CLHEP::HepRotationZ(M_PI / 12),
+                              G4ThreeVector(30, 100, 0));
+
+  [[maybe_unused]] auto vol3PV = new G4PVPlacement(
+      transformVol3, vol3L, "Volume3", vol2L, false, 0, false);
+
+  // Get the lowest volume
+  auto nameSelector =
+      std::make_shared<Acts::Geant4PhysicalVolumeSelectors::NameSelector>(
+          std::vector<std::string>{"olume"}, false);
+
+  Acts::Geant4DetectorSurfaceFactory::Cache cache;
+  Acts::Geant4DetectorSurfaceFactory::Options options;
+  options.sensitiveSurfaceSelector = nameSelector;
+
+  G4Transform3D nominal;
+
+  Acts::Geant4DetectorSurfaceFactory factory;
+  factory.construct(cache, nominal, *worldPV, options);
+
+  auto [element, surface] = cache.sensitiveSurfaces.front();
+  BOOST_CHECK_EQUAL(surface->type(), Acts::Surface::SurfaceType::Plane);
+
+  auto center = surface->center(gctx);
+  auto normal = surface->normal(gctx, center, Acts::Vector3(1, 0, 0));
+  CHECK_CLOSE_ABS(center.x(), 45.981, 1e-3);
+  CHECK_CLOSE_ABS(center.y(), 137.886, 1e-3);
+  CHECK_CLOSE_ABS(center.z(), 144.957, 1e-3);
+  CHECK_CLOSE_ABS(normal.x(), -0.224, 1e-3);
+  CHECK_CLOSE_ABS(normal.y(), 0.592, 1e-3);
+  CHECK_CLOSE_ABS(normal.z(), 0.775, 1e-3);
+
+  Acts::ObjVisualization3D obj;
+  Acts::Vector3 origin(0, 0, 0);
+  Acts::GeometryView3D::drawArrowForward(obj, origin, Acts::Vector3(100, 0, 0),
+                                         1000, 10,
+                                         Acts::ViewConfig({255, 0, 0}));
+  Acts::GeometryView3D::drawArrowForward(obj, origin, Acts::Vector3(0, 100, 0),
+                                         1000, 10,
+                                         Acts::ViewConfig({0, 255, 0}));
+  Acts::GeometryView3D::drawArrowForward(obj, origin, Acts::Vector3(0, 0, 100),
+                                         1000, 10,
+                                         Acts::ViewConfig({0, 0, 255}));
+  Acts::GeometryView3D::drawArrowForward(
+      obj, surface->center(gctx), surface->center(gctx) + 100 * normal, 1000,
+      10, Acts::ViewConfig({0, 255, 0}));
+  int k = 0;
+  for (auto& [el, surf] : cache.sensitiveSurfaces) {
```

Comment:
consider using `Acts::enumerate` or a simple index based for loop

Reply:
side-note: I didn't know we had this helper already. This is great!

Reply:
Done